pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.calcite.rel.hint;
+
+/**
+ * {@code PinotHintOptions} specified the supported hint options by Pinot based a particular type of relation node.
+ *
+ * <p>for each {@link org.apache.calcite.rel.RelNode} type we support a option hint name.</p>
+ * <p>for each option hint name there's a corresponding {@link RelHint} that supported only key-value option stored
+ * in {@link RelHint#kvOptions}</p>
+ */
+public class PinotHintOptions {
+  public static final String AGGREGATE_HINT_OPTIONS = "aggOptions";
+  public static final String JOIN_HINT_OPTIONS = "joinOptions";
+
+  private PinotHintOptions() {
+    // do not instantiate.
+  }
+
+  public static class AggregateOptions {
+    public static final String IS_PARTITIONED_BY_GROUP_BY_KEYS = "partitioned_by_group_by_keys";
+  }
+
+  public static class JoinHintOptions {
+    public static final String JOIN_STRATEGY = "join_strategy";
+    public static final String IS_COLOCATED_BY_JOIN_KEYS = "colocated_by_join_keys";
```

Comment:
Are we planning to keep these hints long-term or are these only until we implement auto-detection for colocation?

Reply:
technically speaking we can keep them --> e.g. even the data are not colocated it doesn't stop the user from "just join those that are colocated on the same server and see the result", it is definitely useful to test the data colocation.